plugins/chat/assets/javascripts/discourse/components/browse-channels.gjs

```diff
@@ -0,0 +1,129 @@
+import i18n from "discourse-common/helpers/i18n";
+import discourseDebounce from "discourse-common/lib/debounce";
+import List from "discourse/plugins/chat/discourse/components/chat/list";
+import ChatModalNewMessage from "discourse/plugins/chat/discourse/components/chat/modal/new-message";
+import ChatChannelCard from "discourse/plugins/chat/discourse/components/chat-channel-card";
+import DcFilterInput from "discourse/plugins/chat/discourse/components/dc-filter-input";
+
+const ARCHIVED = "archived";
+const ALL = "all";
+const OPEN = "open";
+const CLOSED = "closed";
+const TABS = [ALL, OPEN, CLOSED, ARCHIVED];
+
+export default class BrowseChannels extends Component {
+  @service chatApi;
+  @service modal;
+  @service siteSettings;
+
+  @tracked filter = "";
+
+  get currentTab() {
+    return this.args.currentTab ?? ALL;
+  }
+
+  @cached
+  get channelsCollection() {
+    return this.chatApi.channels({
+      filter: this.filter,
+      status: this.currentTab,
+    });
+  }
+
+  get tabs() {
+    if (this.siteSettings.chat_allow_archiving_channels) {
+      return TABS;
+    } else {
+      return [...TABS].removeObject(ARCHIVED);
+    }
+  }
+
+  @action
+  showChatNewMessageModal() {
+    this.modal.show(ChatModalNewMessage);
+  }
+
+  @action
+  setFilter(event) {
+    this.filter = event.target.value;
+    discourseDebounce(this.debouncedLoad, INPUT_DELAY);
+  }
+
+  @action
+  debouncedLoad() {
+    this.channelsCollection.load({ limit: 10 });
+  }
+
+  @action
+  focusFilterInput(input) {
+    schedule("afterRender", () => input?.focus());
+  }
+
+  <template>
+    <div class="chat-browse-view">
+      <div class="chat-browse-view__actions">
+        <nav>
+          <ul class="nav-pills chat-browse-view__filters">
+            {{#each this.tabs as |tab|}}
+              <li class={{concat "chat-browse-view__filter -" tab}}>
+                <LinkTo
+                  @route={{concat "chat.browse." tab}}
+                  class={{concat "chat-browse-view__filter-link -" tab}}
+                  @current-when={{eq tab this.currentTab}}
+                >
+                  {{i18n (concat "chat.browse.filter_" tab)}}
+                </LinkTo>
+              </li>
+            {{/each}}
+          </ul>
+        </nav>
+
+        <DcFilterInput
+          {{didInsert this.focusFilterInput}}
+          @filterAction={{this.setFilter}}
+          @icons={{hash right="search"}}
+          @containerClass="filter-input"
+          placeholder={{i18n "chat.browse.filter_input_placeholder"}}
+        />
+      </div>
+
+      <div class="chat-browse-view__content_wrapper">
+        <div class="chat-browse-view__content">
+          <List
+            @collection={{this.channelsCollection}}
+            class="chat-browse-view__cards"
+            as |list|
+          >
+            <list.Item as |channel|>
+              <ChatChannelCard @channel={{channel}} />
+            </list.Item>
+
+            <list.EmptyState>
```

Comment:
I think we should make the empty state stretch full width, currently follows same layout as list items (ie 50%)

Reply:
Yes I saw it but prefer to not touch it for now